pkg/service/service_test.go

```diff
@@ -339,6 +341,475 @@ func TestPublishClaim(t *testing.T) {
+
+	t.Run("successful publishing the index claim", func(t *testing.T) {
+		mockClaimsService := contentclaims.NewMockContentClaimsService(t)
+		mockProviderIndex := providerindex.NewMockProviderIndex(t)
+		mockBlobIndexLookup := blobindexlookup.NewMockBlobIndexLookup(t)
+		contentLink := testutil.RandomCID()
+
+		ctx := context.Background()
+
+		providerAddr := &peer.AddrInfo{
+			Addrs: []ma.Multiaddr{
+				testutil.Must(ma.NewMultiaddr("/dns/storacha.network/tls/http/http-path/%2Fblobs%2F%7Bblob%7D"))(t),
+				testutil.Must(ma.NewMultiaddr("/dns/storacha.network/tls/http/http-path/%2Fclaims%2F%7Bclaim%7D"))(t),
+			},
+		}
+		// content will have a location claim, an index claim
+		locationDelegationCid, locationDelegation, locationResult := buildTestLocationClaim(t, contentLink.(cidlink.Link), providerAddr)
+		_, indexDelegation, _, indexLink, shardIndex := buildTestIndexClaim(t, contentLink.(cidlink.Link), providerAddr)
+
+		// expect a call to cache the index claim using claims.Publish
+		mockClaimsService.EXPECT().Publish(ctx, indexDelegation).Return(nil)
+
+		// expect a call to find records for index location commitment
+		mockProviderIndex.EXPECT().Find(ctx, providerindex.QueryKey{
+			Hash:         indexLink.Hash(),
+			TargetClaims: []multicodec.Code{metadata.LocationCommitmentID},
+		}).Return([]model.ProviderResult{locationResult}, nil)
+
+		// expect the claim service to be called for each result from providerIndex.Find
+		mockClaimsService.EXPECT().Find(
+			ctx, locationDelegationCid, mock.AnythingOfType("*url.URL"),
+		).Return(locationDelegation, nil)
+
+		// expect the blob index lookup service to be called once to fetch the shard index
+		mockBlobIndexLookup.EXPECT().Find(
+			ctx, mock.Anything, mock.Anything, mock.Anything, mock.Anything,
+		).Return(shardIndex, nil)
+
+		// expect the index claim to be published
+		digests := bytemap.NewByteMap[mh.Multihash, struct{}](-1)
+		for _, slices := range shardIndex.Shards().Iterator() {
+			for d := range slices.Iterator() {
+				digests.Set(d, struct{}{})
+			}
+		}
+		mockProviderIndex.EXPECT().Publish(ctx, *providerAddr, mock.Anything, mock.Anything, mock.Anything).Return(nil)
+
+		err := Publish(ctx, mockBlobIndexLookup, mockClaimsService, mockProviderIndex, *providerAddr, indexDelegation)
+		require.NoError(t, err)
+	})
+
+	t.Run("error when index claim has no capabilities", func(t *testing.T) {
+		mockClaimsService := contentclaims.NewMockContentClaimsService(t)
+		mockProviderIndex := providerindex.NewMockProviderIndex(t)
+		mockBlobIndexLookup := blobindexlookup.NewMockBlobIndexLookup(t)
+
+		ctx := context.Background()
+
+		providerAddr := &peer.AddrInfo{
+			Addrs: []ma.Multiaddr{
+				testutil.Must(ma.NewMultiaddr("/dns/storacha.network/tls/http/http-path/%2Fclaims%2F%7Bclaim%7D"))(t),
+			},
+		}
+
+		// Create a claim with no capabilities
+		claim, err := delegation.Delegate(
+			testutil.Alice,
+			testutil.Bob,
+			[]ucan.Capability[ok.Unit]{}, // No capabilities
```

Comment:
`ok.Unit` is intended to be used for results. It's effectively the same, but it might be clearer to use `ucan.NoCaveats` here.